owncloudApp/src/main/java/com/owncloud/android/MainApp.kt

```diff
@@ -90,14 +91,18 @@ import timber.log.Timber
  * classes
  */
 class MainApp : Application() {
+    private val handlerThreadTag
+        get() = appContext.resources.getString(R.string.ouc_thread_handler_tag)
+    lateinit var handlerThread: HandlerThread
+        private set
```

Comment:
Hi, could you please explain what you mean by this. From my understanding, if you are hinting that the handler thread should be started where it is used then I don't think that would be good, performance wise. Unless I am mistaken. This is Because the handlerThread is being used elsewhere, specifically the `ReceiveExternalFilesActivity.java`. Additionally, I have initialized it inside the MainApp because thread startup takes some time, so it is better to start it up when the application process is started and and make sure it lives (provided the process is alive). So it can be used anytime thus eradicating the startup costs. You can imagine if we started the thread inside the ReceiveExternalFilesActivity.java, that means that everytime the activity is accessed the thread will be started. But I am open to alternative explanation :).